doc/api/buffer.md

```diff
@@ -2622,6 +2623,363 @@ buf.fill(0);
 console.log(buf);
 ```
 
+## Class: TextDecoder
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+An implementation of the [WHATWG Encoding Standard][] `TextDecoder` API.
+
+```js
+const { TextDecoder } = require('buffer');
+const decoder = new TextDecoder('shift_jis');
+let string = '';
+let buffer;
+while(buffer = getNextChunkSomehow()) {
+  string += decoder.decode(buffer, { stream:true });
+}
+string += decoder.decode(); // end-of-stream
+```
+
+### WHATWG Supported Encodings
+
+Per the [WHATWG Encoding Standard][], the encodings supported by the
+`TextDecoder` API are outlined in the table below. For each encoding,
+one or more aliases may be used. Support for some encodings is enabled
+only when Node.js is using the full ICU data.
```

Comment:
https://github.com/nodejs/node/blob/master/BUILDING.md#intl-ecma-402-support
+
https://github.com/nodejs/node/wiki/Intl (linked from the first fragment).